internal/db/repos.go

```diff
@@ -117,6 +117,20 @@ func (s *repos) GetByIDs(ctx context.Context, ids ...api.RepoID) ([]*types.Repo,
 	return s.getReposBySQL(ctx, true, q)
 }
 
+func (s *repos) GetRepoIDsSet(ctx context.Context, ids ...api.RepoID) (map[api.RepoID]*types.Repo, error) {
```

Comment:
Since the returned set is a mapping from `api.RepoID` to `*types.Repo` I think this name makes more sense:

```suggestion
func (s *repos) GetReposSetByIDs(ctx context.Context, ids ...api.RepoID) (map[api.RepoID]*types.Repo, error) {
```

Because we return a set of repos, not a set of repo IDs.